lib/surface/components/form/error_tag.ex

```diff
@@ -124,7 +128,27 @@ defmodule Surface.Components.Form.ErrorTag do
     # Because the error messages we show in our forms and APIs
     # are defined inside Ecto, we need to translate them dynamically.
     Enum.reduce(opts, msg, fn {key, value}, acc ->
-      String.replace(acc, "%{#{key}}", to_string(value))
+      try do
+        String.replace(acc, "%{#{key}}", to_string(value))
+      rescue
+        e ->
+          IO.warn(
+            """
+            the fallback message translator for the `ErrorTag` component cannot handle the given message.
+
+            Hint: you can set up the `default_translator` to route all errors to your application helpers:
+
+              config :surface, :components, [
+                {Surface.Components.Form.ErrorTag, default_translator: {MyAppWeb.ErrorHelpers, :translate_error}}
+              ]
+
+            Original error: #{Exception.message(e)}
```

Comment:
Let's also print the `value`. It might be helpful.

```suggestion
            Given value: #{inspect(value)}
            
            Exception: #{Exception.message(e)}
```